src/bookmarks/background/addition.js

```diff
@@ -50,7 +50,14 @@ export async function createNewPageForBookmark(id, bookmarkInfo) {
     }
 }
 
-export async function createBookmarkByExtension(url) {
+/**
+ * TODO: Decided if we actually need these bookmark docs in Pouch; I don't think they're being used for anything.
+ *
+ * @param {string} url URL to generate page ID from to associate bookmark with.
+ * @throws {Error} Error thrown if page for supplied URL not indexed.
+ * @returns {Promise<void>}
+ */
+export async function createBookmarkByUrl(url) {
     const pageId = generatePageDocId({ url })
     const pageDoc = await db.get(pageId)
```

Comment:
Now popup bookmarking doesn't work at all until the visit log happens because this will throw an error as the page doesn't yet exist. @mukeshkharita Didn't you have the logic before that switched to making the page doc if it's not yet ready? Where did that go?